### DISCUSSION TITLE
How to make __name__ to be "__main__" in Kyuubi python operation?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#6817] [Q&A]
I find __ name__  is "builtins" but not "__ main__" when running a Kyuubi python operation on Spark. 
What is the cause of this?
And is there a way to make it "__ main__"?

**Comment by HaoYang670** [comment]
HI @pan3793 Do you have any insight?

**Comment by pan3793** [comment]
I'm not familiar with PySpark, cc @turboFei @cfmcgrady @iodone

**Comment by turboFei** [comment]
cc @cfmcgrady

**Comment by turboFei** [comment]
I do not understand this question, could you provide some screenshot to help to understand it? @HaoYang670

**Reply by HaoYang670** [reply]
When I submit a Kyuubi Python operation with code "print(__ name__)", it returns me "builtins", but not "__ main__".

I want to make it "__ main__" to run Python in the format of "if __ name__ == "__ main__": "

**Reply by turboFei** [reply]
I think the reason is that, all the codes are running in the kyuubi python interpreter.
https://github.com/apache/kyuubi/blob/27c734ed9555931ec9970e9cecedfabbf8ea8a93/externals/kyuubi-spark-sql-engine/src/main/resources/python/execute_python.py#L545-L546

**Reply by HaoYang670** [reply]
Is there way to change the value of it?

**Reply by turboFei** [reply]
There is no reason to change it in the community kyuubi.

**Reply by HaoYang670** [reply]
Or, any configuration I can set it? Or could you help to point it out where to modify it in the source code, we may need to build a specific version of Kyuubi for our internal usage. Thanks

**Reply by turboFei** [reply]
not sure, I am not familiar with python. cc @cfmcgrady

**Reply by HaoYang670** [reply]
@cfmcgrady is there any insight? Thanks
